src/Nethermind/Nethermind.Merge.Plugin/Data/BlobAndProofV1.cs

```diff
@@ -0,0 +1,24 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using System;
+using Nethermind.Core;
+
+namespace Nethermind.Merge.Plugin.Data;
+
+public class BlobAndProofV1
+{
+    public BlobAndProofV1(Transaction blobTx, int index)
+    {
+        if (blobTx is not { NetworkWrapper: ShardBlobNetworkWrapper wrapper })
+        {
+            throw new ArgumentException("Shard blob transaction should contain network wrapper data");
+        }
+
+        Blob = wrapper.Blobs[index];
+        Proof = wrapper.Proofs[index];
+    }
```

Comment:
IMO transaction passed here and validation here is bad design, it should happen elsewhere - probably in the tx pool, and maybe we need to load blobs from storage then.